src/cli/args.rs

```diff
@@ -93,8 +94,159 @@ pub(crate) enum CliCommands {
 
     /// List the current checkpoints of the db.
     ListCheckpoints {},
+
+    /// Runs a garbage collection for a specific resource type once
+    RunGarbageCollection {
+        /// the type of resource to clean up (manifest, wal, compacted)
+        #[arg(short, long)]
+        resource: GcResource,
+
+        /// the minimum age of the resource before considering it for GC
+        #[arg(short, long)]
+        #[clap(value_parser = humantime::parse_duration)]
+        min_age: Duration,
+    },
+
+    /// Schedules a period garbage collection job
+    #[command(group(
+    ArgGroup::new("gc_config")
+        .args(["manifest", "wal", "compacted"])
+        .multiple(true)
+        .required(true)
+    ))]
+    ScheduleGarbageCollection {
+        /// Configuration for manifest garbage collection should be set in the
+        /// format min_age=<duration>,period=<duration> -- the min_age is the
+        /// minimum manifest age that should be considered for collection and
+        /// the period is how often to attempt a GC
+        #[arg(long, value_parser = parse_gc_schedule)]
+        manifest: Option<GcSchedule>,
+
+        /// Configuration for WAL garbage collection should be set in the
+        /// format min_age=<duration>,period=<duration> -- the min_age is the
+        /// minimum WAL age that should be considered for collection and
+        /// the period is how often to attempt a GC
+        #[arg(long, value_parser = parse_gc_schedule)]
+        wal: Option<GcSchedule>,
+
+        /// Configuration for compacted SST garbage collection should be set in the
+        /// format min_age=<duration>,period=<duration> -- the min_age is the
+        /// minimum SST age that should be considered for collection and
+        /// the period is how often to attempt a GC
+        #[arg(long, value_parser = parse_gc_schedule)]
+        compacted: Option<GcSchedule>,
+    },
+}
+
+#[derive(Debug, Clone, ValueEnum)]
+pub(crate) enum GcResource {
+    Manifest,
+    Wal,
+    Compacted,
+}
+
+fn parse_gc_schedule(s: &str) -> Result<GcSchedule, String> {
+    let parts: HashMap<String, String> = s
+        .split(',')
+        .filter_map(|kv| {
+            let mut parts = kv.splitn(2, '=');
+            match (parts.next(), parts.next()) {
+                (Some(key), Some(value)) => Some((key.to_string(), value.to_string())),
+                _ => None,
+            }
+        })
+        .collect();
+
+    let min_age = parts
+        .get("min_age")
+        .ok_or_else(|| "Missing or invalid 'min_age'".to_string())
+        .and_then(|v| {
+            humantime::parse_duration(v).map_err(|e| {
+                "Could not parse min_age as duration: "
+                    .to_string()
+                    .to_owned()
+                    + e.to_string().as_str()
+            })
+        })?;
+    let period = parts
```

Comment:
it's two different CLIs, the `RunGarbageCollection` CLI does just run once without a period:
```
    /// Runs a garbage collection for a specific resource type once
    RunGarbageCollection {
        /// the type of resource to clean up (manifest, wal, compacted)
        #[arg(short, long)]
        resource: GcResource,

        /// the minimum age of the resource before considering it for GC
        #[arg(short, long)]
        #[clap(value_parser = humantime::parse_duration)]
        min_age: Duration,
    },
```

Reply:
I split it up that way because it makes each individual command much easier to reason about (and they're probably used in different scenarios anyway, I don't imagine you'd want to kick off a scheduled compaction for WALs but a one-off compaction for Manifests)